### ISSUE TITLE
Search result links should be stable versions if possible

### ISSUE THREAD

**Issue by loic-sharma** [NuGet/NuGetGallery#8742]
### Related Problem

nuget.org's search results link to the display package page to browse a package's details. Today, this links to the latest version (including pre-release versions). This is not ideal as pre-release versions may have bugs, and updates may introduce breaking changes. Generally, the latest stable version is a "safer" choice.

### The Elevator Pitch

nuget.org search results should link to the latest stable version, if one exists. Only if a package has no stable versions should the search results link to a pre-release version.

**Comment by joelverhagen** [comment]
This seems like a pretty safe change. If @jcjiang likes it, let's ship it with a feature flag. Think how we can measure success.

**Comment by jcjiang** [comment]
Loic and I got a poll up on Twitter, here are the results so far:

![image](https://user-images.githubusercontent.com/23298667/129779452-b4295757-cf76-400a-a7cf-41cdf1684d08.png)

So far looks like users want the change! Will give the poll some more time, we're seeing individual feedback preferring prerelease so it would be good to understand why. Otherwise I think it's safe to ship it with a feature flag

**Comment by loic-sharma** [comment]
Here are the final results: https://twitter.com/jiachenjiang_/status/1427693050222891008

![image](https://user-images.githubusercontent.com/737941/130108702-49680b53-49b4-44cd-ae82-b2d12e94e2cf.png)

201 votes, ~90% preferred links to stable versions

**Comment by jcjiang** [comment]
Seems like an overwhelming number of users prefer being directed to the
latest stable version. Loic and I had been talking about whether this
behavior should happen in the VS client as well.

Thoughts on sharing the survey result with the client team and suggest
aligning with this behavior?

**Comment by loic-sharma** [comment]
@jcjiang Yup please share the results with the client team. Do note that Visual Studio has a slower update cadence than nuget.org. It may make sense to try this change on nuget.org first to see what feedback we get. If there's favorable (or no) feedback, we could go ahead and update Visual Studio as well.

**Comment by augustoproiete** [comment]
Just to add to the conversation, LINQPad by default shows stable versions only, unless the user clicks the "Include Prerelease" checkbox.

![image](https://user-images.githubusercontent.com/177608/130146263-df7073ed-0db3-4bd2-b5c6-440eff5f4c4c.png)

**Comment by loic-sharma** [comment]
Thanks for the input! This is indeed a third option we're considering. On nuget.org the "Include prelease" option is tucked away in the "Filters" drawer. This can be missed pretty easily, and we're concerned that folks may think packages are "missing" if we exclude pre-release by default. We'll continue to iterate on user feedback though :)
